tools/run_tests/xds_k8s_test_driver/framework/helpers/grpc.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2023 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This contains common helpers for working with grpc data structures."""
+import functools
+from typing import Optional
+
+import grpc
+
+
+@functools.cache  # pylint: disable=no-member
```

Comment:
Nice usage, I'm stealing it for the PR I'm working on :)

Reply:
Note this is only available since 3.9. Before that you need to do the usual `@functools.lru_cache(maxsize=None)`